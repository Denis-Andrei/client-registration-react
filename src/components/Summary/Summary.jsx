import React from 'react'
import { Table, BackLink, Button } from 'govuk-react'
import { useHistory } from 'react-router-dom';
import { UserProperties } from '../../utils/UserProperties';
import { getData } from '../../utils/Utils';


export default function Summary() {
    const history = useHistory();
    
    const objToBeSend = {
        "name": getData(UserProperties.name),
        "businessName": getData(UserProperties.businessName),
        "contactNumber": getData(UserProperties.contactNumber),
        "propertyNumber": getData(UserProperties.propertyNumber),
        "postcode": getData(UserProperties.postcode),
        "businessType": getData(UserProperties.businessType),
        "password": getData(UserProperties.password),
    }
    const requestOptions = {  
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            "Content-Type": "application/json",
        },
        body: JSON.stringify(objToBeSend),
    }
    
    const registerUser = () => {
        fetch("/clients/register", requestOptions) 
        .then((res) => {
            if(res.status === 200){
                return res.json();
            }
        })
        .then(data =>{
            sessionStorage.setItem(UserProperties.crn, data.crn)
            history.push('/success')
        })
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        registerUser()
    }
    return (
        <div>
            <BackLink data-testid="goBack" onClick={ () => history.goBack()}>Back</BackLink>
            <Table caption="Summary">
                <Table.Row>
                    <Table.Cell data-testid="table-cell-name">Name</Table.Cell>
                    <Table.Cell>{getData(UserProperties.name)}</Table.Cell>
                </Table.Row>
                <Table.Row>
                    <Table.Cell data-testid="table-cell-business">Business Name</Table.Cell>
                    <Table.Cell>{getData(UserProperties.businessName)}</Table.Cell>
                </Table.Row>
                <Table.Row>
                    <Table.Cell data-testid="table-cell-number">Contact Number</Table.Cell>
                    <Table.Cell>{getData(UserProperties.contactNumber)}</Table.Cell>
                </Table.Row>
                <Table.Row>
                    <Table.Cell data-testid="table-cell-address">Address</Table.Cell>
                    <Table.Cell>{getData(UserProperties.propertyNumber)}, {getData(UserProperties.postcode)}</Table.Cell>
                </Table.Row>
                <Table.Row>
                    <Table.Cell data-testid="table-cell-business-type">Business Type</Table.Cell>
                    <Table.Cell>{getData(UserProperties.businessType)}</Table.Cell>
                </Table.Row>
            </Table>

            <Button data-testid="submit" onClick={handleSubmit}>Submit</Button>
        </div>
    )
}
