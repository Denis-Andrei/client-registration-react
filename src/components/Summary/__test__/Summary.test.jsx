import React from 'react';
import ReactDOM from 'react-dom';
import Summary from '../Summary';
import { render, cleanup, fireEvent } from '@testing-library/react'
import renderer from "react-test-renderer"

afterEach(cleanup);

const mockHistoryPush = jest.fn();
const mockGoBack = jest.fn();


jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        push: mockHistoryPush,
        goBack: mockGoBack
    }),
}));



it("Summary() renders withoud crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Summary />, div)
})

it("renders Summary() correctly", () => {
    const { getByTestId }  = render(<Summary />)
    expect(getByTestId('table-cell-name')).toHaveTextContent("Name")
    expect(getByTestId('table-cell-business')).toHaveTextContent("Business Name")
    expect(getByTestId('table-cell-number')).toHaveTextContent("Contact Number")
    expect(getByTestId('table-cell-address')).toHaveTextContent("Address")
    expect(getByTestId('table-cell-business-type')).toHaveTextContent("Business Type")
})

it("matches snapshot", () => {
    const tree = renderer.create(<Summary />).toJSON();
    expect(tree).toMatchSnapshot();
})


// describe("handleSubmit() ", () => {
//     it("onSubmit redirects to the correct route", () => {
        
//         const { getByTestId } = render(<Summary />)
//         const submit = getByTestId("submit");
       
//         const mockRegisterUser =  jest.fn()
//         Utils.registerUser = mockRegisterUser

//         fireEvent.click(submit);
//         expect(mockRegisterUser).toHaveBeenCalled();
//     })
// })

describe("goBack() ", () => {
    it("should be called on click event", () => {
        const { getByTestId } = render(<Summary />)
        const goBackBtn = getByTestId("goBack");
    
        fireEvent.click(goBackBtn)
        expect(mockGoBack).toHaveBeenCalled();
        
    })
})

