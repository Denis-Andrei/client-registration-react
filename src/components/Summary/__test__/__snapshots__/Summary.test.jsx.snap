// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <a
    className="src__Anchor-sc-11edswm-0 gGueb"
    data-testid="goBack"
    onClick={[Function]}
  >
    Back
  </a>
  <table
    className="src__StyledTable-sc-1o56d04-0 glPCiB"
  >
    <caption
      className="Caption-sc-1idz0wn-0 ehgwzr"
    >
      Summary
    </caption>
    <tbody
      className="TableBody-sc-1gk0z9f-0"
    >
      <tr
        className="Row-sc-17ofjw0-0"
      >
        <td
          className="Cell-e1qzac-0 eAlzkC"
          data-testid="table-cell-name"
        >
          Name
        </td>
        <td
          className="Cell-e1qzac-0 eAlzkC"
        >
          
        </td>
      </tr>
      <tr
        className="Row-sc-17ofjw0-0"
      >
        <td
          className="Cell-e1qzac-0 eAlzkC"
          data-testid="table-cell-business"
        >
          Business Name
        </td>
        <td
          className="Cell-e1qzac-0 eAlzkC"
        >
          
        </td>
      </tr>
      <tr
        className="Row-sc-17ofjw0-0"
      >
        <td
          className="Cell-e1qzac-0 eAlzkC"
          data-testid="table-cell-number"
        >
          Contact Number
        </td>
        <td
          className="Cell-e1qzac-0 eAlzkC"
        >
          
        </td>
      </tr>
      <tr
        className="Row-sc-17ofjw0-0"
      >
        <td
          className="Cell-e1qzac-0 eAlzkC"
          data-testid="table-cell-address"
        >
          Address
        </td>
        <td
          className="Cell-e1qzac-0 eAlzkC"
        >
          
          , 
          
        </td>
      </tr>
      <tr
        className="Row-sc-17ofjw0-0"
      >
        <td
          className="Cell-e1qzac-0 eAlzkC"
          data-testid="table-cell-business-type"
        >
          Business Type
        </td>
        <td
          className="Cell-e1qzac-0 eAlzkC"
        >
          
        </td>
      </tr>
    </tbody>
  </table>
  <button
    className="sc-bdnxRM gXVtdA"
    data-testid="submit"
    disabled={false}
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
